File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        lastSeen = {}

        for idx, num in enumerate(nums):
            if target - num in lastSeen:
                return [idx, lastSeen[target - num]]
            else:
                lastSeen[num] = idx
        
        return

File: 0001-two-sum/0001-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var numToIdx: [Int:Int] = [:]

        for (idx, val) in nums.enumerated() {
            if let compleIdx = numToIdx[target - val] {
                return [idx, compleIdx]
            } 

            numToIdx[val] = idx
            
        }

        return []
    }
}